functests/vrf/vrf.go

```diff
@@ -0,0 +1,338 @@
+					[]map[string]string{
+						{"vrfName": VRFBlueName, "vrfClinetIP": podClientVRFBlueIPAddress, "vrfInterface": "net1"},
+						{"vrfName": VRFRedName, "vrfClinetIP": podClinetVRFRedOverlappingIP, "vrfInterface": "net2"}})
+				Expect(err).ToNot(HaveOccurred())
+				err = podHasCorrectVrfConfig(apiclient, PodServer.Name,
+					[]map[string]string{
+						{"vrfName": VRFBlueName, "vrfClinetIP": podServerVRFBlueIPAddress, "vrfInterface": "net1"},
+						{"vrfName": VRFRedName, "vrfClinetIP": podServerVRFRedOverlappingIP, "vrfInterface": "net2"}})
+				Expect(err).ToNot(HaveOccurred())
+				err = validateVRFConnectivity(apiclient, PodClient.Name, VRFRedName, podServerVRFRedOverlappingIP)
+				Expect(err).ToNot(HaveOccurred())
+				err = validateVRFConnectivity(apiclient, PodClient.Name, VRFBlueName, podServerVRFBlueIPAddress)
+				Expect(err).ToNot(HaveOccurred())
+				err = apiclient.Pods(TestNamespace).Delete(context.Background(), PodServer.Name, metav1.DeleteOptions{GracePeriodSeconds: pointer.Int64Ptr(0)})
+				Expect(err).ToNot(HaveOccurred())
+				Eventually(func() error {
+					_, err := apiclient.Pods(TestNamespace).Get(context.Background(), PodServer.Name, metav1.GetOptions{})
+					return err
+				}, podWaitingTime, 5*time.Second).Should(HaveOccurred())
+
+				err = validateVRFConnectivity(apiclient, PodClient.Name, VRFBlueName, podServerVRFBlueIPAddress)
+				Expect(err).To(HaveOccurred())
+				err = validateVRFConnectivity(apiclient, PodClient.Name, VRFRedName, podServerVRFRedOverlappingIP)
+				Expect(err).To(HaveOccurred())
+				err = validateVRFConnectivity(apiclient, PodClient.Name, "eth0", podServerVRFRedOverlappingIP)
+				Expect(err).ToNot(HaveOccurred())
+			},
+			Entry(describe(""), sameNode, ipStackIPv4),
+			Entry(describe(""), diffNode, ipStackIPv4),
+		)
+	})
+})
+
+func validateVRFConnectivity(cs *apiclient.ClientSet, client string, vrfName string, DestIPAddr string) error {
+	pingCommand := []string{"ping", "-I", vrfName, "-c5", DestIPAddr}
+	pod, err := cs.Pods(TestNamespace).Get(context.Background(), client, metav1.GetOptions{})
+	pingStatus, err := pods.ExecCommand(cs, *pod, pingCommand)
+	if err != nil {
+		return err
+	}
+	if strings.Contains(pingStatus.String(), " 0% packet loss") {
+		return nil
+	}
+	return fmt.Errorf("Connectivity test error")
+}
+
+func addVrfNad(NadName string, ifName string, vrfName string) netattdefv1.NetworkAttachmentDefinition {
+	return netattdefv1.NetworkAttachmentDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: NadName,
+			Namespace:    TestNamespace,
+		},
+		Spec: netattdefv1.NetworkAttachmentDefinitionSpec{
+			Config: fmt.Sprintf(`{"cniVersion": "0.4.0", "name": "macvlan-vrf", "plugins": [{"type": "macvlan","master": "%s","ipam": {"type": "static"}},{"type": "vrf","vrfname": "%s"}]}`, ifName, vrfName),
+		},
+	}
+}
+
+func definePodwithVrfStaticIP(name string, namespace string, nodeName string, networks string) *k8sv1.Pod {
```

Comment:
This is just a pod with network.
Can you use / refactor existing functions? And in any case, I'd see this to be in the pods package